grpc/eth.go

```diff
@@ -0,0 +1,45 @@
+package grpc
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/RTradeLtd/config"
+	"github.com/RTradeLtd/grpc/temporal/request"
+	"github.com/RTradeLtd/grpc/temporal/response"
+	pb "github.com/RTradeLtd/grpc/temporal/service"
+	"google.golang.org/grpc"
+)
+
+// Client is how we interface with the GRPC server as a client
+type Client struct {
+	GC *grpc.ClientConn
+	SC pb.SignerClient
+}
+
+// NewGRPCClient generates our GRPC API client
+func NewGRPCClient(cfg *config.TemporalConfig, insecure bool) (*Client, error) {
+	grpcAPI := fmt.Sprintf("%s:%s", cfg.API.Payment.Address, cfg.API.Payment.Port)
+	var (
+		gconn *grpc.ClientConn
+		err   error
+	)
+	if insecure {
+		gconn, err = grpc.Dial(grpcAPI, grpc.WithInsecure())
+	}
+	if err != nil {
+		return nil, err
+	}
+	sconn := pb.NewSignerClient(gconn)
```

Comment:
While this might be an alright start, we probably need to:

- set up TLS
- set up some client-server authentication mechanism

before this goes live

Reply:
Agreed, tracked in TEM-110